internal/interface/handlers/document_handler.go

```diff
@@ -107,7 +108,32 @@ func (h *DocumentHandler) CreateDocument(c *fiber.Ctx) error {
 // @Failure 500 {object} response.Response
 // @Router /documents/{document_id} [put]
 func (h *DocumentHandler) UpdateDocumentByID(c *fiber.Ctx) error {
-	return nil
+	documentID := c.Params("document_id")
+	if documentID == "" {
+		resp := response.NewResponseFactory(response.ERROR, "Document ID is required")
+		return resp.SendResponse(c, fiber.StatusBadRequest)
+	}
+
+	// Parse the request body into UpdateDocumentDTO
+	var updateDocumentDTO dtos.UpdateDocumentDTO
+	if err := c.BodyParser(&updateDocumentDTO); err != nil {
+		resp := response.NewResponseFactory(response.ERROR, "Invalid request body")
+		return resp.SendResponse(c, fiber.StatusBadRequest)
+	}
+
+	err := h.documentUsecase.UpdateDocumentByID(documentID, updateDocumentDTO)
+	if err != nil {
+		if err.Error() == "document not found" {
+			resp := response.NewResponseFactory(response.ERROR, "Document not found")
+			return resp.SendResponse(c, fiber.StatusNotFound)
+		}
+		resp := response.NewResponseFactory(response.ERROR, "Failed to update document")
+		return resp.SendResponse(c, fiber.StatusInternalServerError)
+	}
```

Comment:
you can use this instead so that the logic for checking status code is all in usecase layer
```go
apperr := h.documentUsecase.UpdateDocumentByID(documentID, updateDocumentDTO)
	if apperr != nil {
		resp := response.NewResponseFactory(response.ERROR, apperr.Error())
		return resp.SendResponse(c, apperr.HttpCode)
	}
```